src/dali/plugin/input/csv/blockfi.py

```diff
@@ -133,9 +177,98 @@ def load(self) -> List[AbstractTransaction]:
                             crypto_received=Keyword.UNKNOWN.value,
                         )
                     )
-                elif line[self.__TYPE_INDEX] == _WITHDRAWAL_FEE:
+                elif entry_type == _ACH_DEPOSIT:
+                    last_withdrawal_fee = None
+                    result.append(
+                        InTransaction(
+                            plugin=self.__BLOCKFI,
+                            unique_id=Keyword.UNKNOWN.value,
+                            raw_data=raw_data,
+                            timestamp=f"{line[self.__TIMESTAMP_INDEX]} -00:00",
+                            asset=line[self.__CURRENCY_INDEX],
+                            exchange=self.__BLOCKFI,
+                            holder=self.account_holder,
+                            transaction_type="Buy",
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_in=line[self.__AMOUNT_INDEX],
+                            fiat_fee="0",
+                            notes="Ach deposit",
+                        )
+                    )
+                elif entry_type == _TRADE:
+                    # Trades will be handled by parsing trade_report_all.csv
+                    # export
+                    continue
+                elif entry_type == _WITHDRAWAL_FEE:
                     last_withdrawal_fee = RP2Decimal(line[self.__AMOUNT_INDEX])
                     last_withdrawal_fee = -last_withdrawal_fee  # type: ignore
                 else:
-                    self.__logger.debug("Unsupported transaction type (skipping): %s", raw_data)
+                    self.__logger.error("Unsupported transaction type (skipping): %s", raw_data)
+
+        if self.__trade_csv_file:
+            result += self.parse_trade_report(self.__trade_csv_file)
+        return result
+
+    def parse_trade_report(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(file_path, mode="r", encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+            # skip csv header
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            col_idx = {}
+            for (idx, name) in enumerate(header):
+                col_idx[name] = idx
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                trans_type = line[col_idx["Type"]]
+                if trans_type != "Trade":
+                    raise Exception(f"Internal error: unsupported transaction type: {trans_type}")
+
+                trade_id = line[col_idx["Trade ID"]]
+                date = line[col_idx["Date"]]
+                timestamp = f"{date} -00:00"
```

Comment:
Type hints

Reply:
BTW, I think a lot of the type hints in this project are not needed because mypy does type inference like most modern programming languages. Adding unnecessary type hints does make the code less readable compared to other idiomatic python projects. I am usually the guy who asks people to add type annotations in python code base during code reviews, but I think we are over doing it in rp2 :P

Reply:
For these couple lines, mypy should have the info to make the inference that these variables are all strings.

Reply:
I would disagree on the code being less readable with type hints: yes, mypy does type inference but type hints make type declaration explicit (vs implicit with inference), which to me makes the code clearer. Also by adding types everywhere we remove the arbitrary decision of when to add types vs not. But I don't want to make this into a big thing for such a small snippet: it's ok to leave them out :-)

Reply:
> Also by adding types everywhere we remove the arbitrary decision of when to add types vs not. 

I agree this is a useful rule to establish to make code reviews more efficient for both sides. The rule I usually apply is to make sure function/method signatures are annotated and only annotate the rest of the code base enough to pass mypy. Occasionally, I will annotate a variable if it really helps readability, but usually well written code is self-explanatory and so this doesn't happen often. Type annotations are mostly for compilers, humans usually can figure the types out with good variable/function names plus context. For example when I read a variable named `transaction_list`, I would implicitly assume it to be a list of transactions. So to me, `transaction_list: List[AbstractTransaction]` makes the code more verbose and hard to read. Same for `amount: RP2Decimal = RP2Decimal(transaction[_AMOUNT][_AMOUNT])`, the type for the amount variable should be obvious for a human and a machine. IMHO, redundant info hurts readability. It's certainly not an easy balance to strike because lack of essential info also hurts readability.

I assume you work with a large monitor and editor window. For smaller windowses, excessive type annotations plus long variable names make things worse because many times I won't be able to see the full line without performing horizontal scrolling in my editor. Human eyes are better at scanning lines vertically instead of horizontally. Adding horizontal window scroll makes it even worse. Here is a screenshot of what it looks like in my coding environment:

![vim-screen-shot](https://user-images.githubusercontent.com/670302/160262177-7b976904-327a-408c-9f4a-ae9c14903153.png)

This is why the official python style guide recommends a max line length of 79 chars: https://peps.python.org/pep-0008/#maximum-line-length. I try to not be doctrinal about this because there are rare cases where going pass that 79 limit improves readability, but I think in general, it's a good practice to keep lines short.

Something is not right if the python code is a lot more verbose than the Go and Rust code I write these days ;)

Reply:
I have used strongly typed languages for a long time and am of the school of thought that explicit types are a good thing: it's a zero-tolerance approach against potential ambiguity, even when the ambiguity is little or none (again, removing the decision is the main point). I have lived through many horror stories, where compiler warnings were disabled and inexplicable bugs occurred due to uncaught typing errors (but in fairness that was with C, not Python, so it's a bit apples/oranges). In this case I guess delegating the decision to mypy is an acceptable compromise.

As for monitors, I do have a large monitor and imagine many coders do (the price of monitors has been falling for many years, reducing the barrier of entry to getting a large one). I always thought the 80 character limit thing was a legacy from the 80s that never died, but I admit I like your point about human eyes being better at scanning vertically. :)